# Crosstalk between codon optimality and cis-regulatory elements dictates mRNA stability

This repository contains data and scripts for the analysis found in "**Crosstalk between codon optimality and cis-regulatory elements dictates mRNA stability**" paper.

by *Santiago Gerardo Medina-Muñoz, Gopal Kushawah, Luciana Andrea Castellano, Michay Diez, Michelle Lynn DeVore, María José Blanco Salazar, Ariel A Bazzini*

[![DOI](https://zenodo.org/badge/164946878.svg)](https://zenodo.org/badge/latestdoi/164946878)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)


### Abstract

**Background:** The regulation of messenger RNA (mRNA) stability has a profound impact on gene expression dynamics during embryogenesis. For example, in animals, maternally deposited mRNAs are degraded after fertilization to enable new developmental trajectories. Regulatory sequences in 3′ untranslated regions (3′UTRs) have long been considered the central determinants of mRNA stability. However, recent work indicates that the coding sequence also possesses regulatory information. Specifically, translation in cis impacts mRNA stability in a codon-dependent manner. However, the strength of this mechanism during embryogenesis, as well as its relationship with other known regulatory elements, such as microRNA, remains unclear.

**Results:** Here, we show that codon composition is a major predictor of mRNA stability in the early embryo. We show that this mechanism works in combination with other cisregulatory elements to dictate mRNA stability in zebrafish and Xenopus embryos as well as in mouse and human cells. Furthermore, we show that microRNA targeting efficacy can be affected by substantial enrichment of optimal (stabilizing) or non-optimal (destabilizing) codons. Lastly, we find that one microRNA, miR-430, antagonizes the stabilizing effect of optimal codons during early embryogenesis in zebrafish.

**Conclusions:** By integrating the contributions of different regulatory mechanisms, our work provides a framework for understanding how combinatorial control of mRNA stability shapes the gene expression landscape.


### Project Organization

This project is organized as suggested by [Nobble 2009](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000424).

#### [data |](data/) the data-sets of the project.

Here, you can find some the data sets used in the analysis. For published data go to the folder: [data/00-00-00-PublishedData/](/data/00-00-00-PublishedData). For a detailed description of each dataset check the readme file: [data/00-00-00-PublishedData/README.md](/data/00-00-00-PublishedData/README.md).


#### [Figures-Paper.md |](Figures-Paper.md) the R code to replicate the figures in the publication.


#### [results |](results/) the computational analyses that were performed, each analysis is a folder named sorted chronologically

- Some analyses presented here were preliminary and are not part of the final publication.


#### [paper-analysis |](paper-analysis/) additional computational analyses

- These analyses are most of the final analysis presented in the paper. Most of the code here can be used to reproduce the figures in the paper.

**Note:** Most of the analysis and scripts are self-documented.

#### [doc |](doc/) Some preliminary notes


#### Questions/contact

If you have questions regarding the analysis or data you can send me an email at: `santiago.medina@cinvestav.mx`
