---
title: "Decay Rate Estimation"
author: "Santiago Medina"
date: "1/11/2019"
output:
   html_document:
    keep_md: true
---

### Overview:

The goal of this analysis is to estimate the decay rate for the endogenous genes with the a-amanitin data.

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = F, message = F, warning = F, collapse = T,
  fig.path = "./figures/", dev = c("png", "pdf"),
  dpi=300
  )

# libraries need it for analysis
library(tidyverse)
library(zFPKM)
library(broom)
library(brms)

theme_set(theme_light())
```

### 1. Tidy data for analysis

I use the a-amanitin data set. This includes 12 time points.

```{r data}
amanitin <- read_csv("../../data/19-01-09-RNAseqProfilesFish/rna-seq-profiles/alpha-amanitin-prolife.csv")

# reshape data in more usefull format
time_course <- 
  amanitin %>% 
  select(-Name) %>% 
  gather(key = sample, value = TPM, -Gene_ID) %>% 
  mutate(
    time = str_extract(sample, pattern = "\\d\\.?\\d?") %>% as.numeric()
  ) %>% 
  select(-sample)

```

### 2. Detect active genes

To estimate the decay rate I will use only active genes. I will use the 
[zFPKM](https://bioconductor.org/packages/release/bioc/html/zFPKM.html) package for this.

Plot of the distribution of gene expression.

```{r expression_distribution}
time_course %>% 
  ggplot(aes(x = TPM, color = time, group = time)) +
  geom_density() +
  scale_x_log10() +
  scale_color_viridis_c()
```

as the authors suggest a cut-off of zFPKM > -3 is used.


```{r active_genes}
zfpkm <- 
  amanitin %>% 
  select(-Name, -Gene_ID) %>% 
  as.data.frame() %>% 
  zFPKM(assayName = "tpm")

active_genes <- 
  amanitin[rowMeans(zfpkm) > -3, ] %>% 
  pull(Gene_ID)

time_course <- filter(time_course, Gene_ID %in% active_genes)

time_course %>% 
  ggplot(aes(x = TPM, color = time, group = time)) +
  geom_density() +
  scale_x_log10() +
  scale_color_viridis_c() +
  labs(
    titlte = "TPM Distribuiton for active genes"
  )

## save a copy of the active and tidy genes (time_cours)

write_csv(time_course, "results_data/aamanitin_time_course_tidy_only_active_genes.csv")
```

### 3. Estimate Decay Rate

Now, I display the time course for a set of random genes.

```{r}
## 
set.seed(123)
s_genes <- time_course$Gene_ID %>% 
  sample(12)

time_course %>% 
  filter(Gene_ID %in% s_genes) %>% 
  ggplot(aes(x = time, y = TPM)) +
  geom_point() +
  facet_wrap(~Gene_ID, scales = "free") +
  scale_y_log10() + 
  geom_smooth(method = lm) +
  labs(
    title = "RNA-seq time course",
    subtitle = "A group of random genes is displayed",
    x = "time (hrs)"
    )
```

#### Model

To estimate the decay rate I will use the following linear model

$$
\begin{aligned}
logTPM &\sim Normal(\mu_i, \sigma) \\
\mu_i & = \alpha + \beta\;time
\end{aligned}
$$
Note: I subtract -2 from the time variable, so the $\alpha$ parameter has the interpretation of the RNA-level at time 2 hrs.


```{r get_estimates}

fit_unpooled <- 
  time_course %>% 
  group_by(Gene_ID) %>% 
  mutate(time = time - min(time)) %>% # subtract time so it starts at 0
  nest() %>% 
  mutate(
    fit = map(data, ~lm(log(TPM) ~ time, data = .)),
    tidy_fit = map(fit, tidy, conf.int = T),
    r2_score = map(fit, glance)
  )

# save results

fit_unpooled %>% 
  unnest(tidy_fit) %>% 
  mutate(
    term = str_replace(term, pattern = "\\(Intercept\\)", "alpha"),
    term = str_replace(term, pattern = "time", "beta"),
  ) %>% 
  write_csv("results_data/estimated_decay_rates.csv")
  

x1 <- fit_unpooled %>% 
  unnest(tidy_fit) %>% 
  select(Gene_ID:estimate) %>% 
  spread(key = term, value = estimate)

x1 %>% 
  ggplot(aes(x=`(Intercept)`, y=time)) +
  geom_point(size = 1/4) +
  geom_smooth() +
  labs(
    x = "alpha (log Expression at initial time point)",
    y = "beta (decay rate)",
    title = "bi-variate distribution of model parameters"
  ) +
  geom_vline(xintercept = 0, linetype=3, color = "red")

```

Note that to the left 0 (expression) the genes tend to have a positive $\beta$, this maybe just noise.


***
### Session Info

```{r session}
sessionInfo()
```

