# January

```{r settings, echo=FALSE}
knitr::opts_chunk$set(echo = F)
```


## 19-01-09

+ Created directory organization based on [@10.1371/journal.pcbi.1000424]
+ Started version control with git. I host a private [github repo](https://github.com/santiago1234/MZT-rna-stability)
+ Downloaded mRNA seq time courses generated by the lab.

I have to re-run previews analysis and do some quality checks on this data. Ariel suggested to do: PCA, poly-A vs ribo (also the ratio) and then the model.


## 19-01-10

+ Run Exploratory analysis for RNA-seq time course (PCA), the PCA analysis looks find.

I also generated the following bi-variate distribution:

```{r bivariate, out.width='130%', fig.cap='time course distribution (TPM)'}
knitr::include_graphics("../../results/19-01-10-EDA-RNA-time-course/figures/bivariate_polyA-1.png")
```


- For the alpha-amanitin-time course I have to see what is the best way to perform the SPIKE-in normalization, I can
  ask Ariel or Comp-Bio. Also the following paper [@risso2014normalization] can be useful to read.


## 19-01-15 

#### Estimating decay rate a-amanitin data

Today I focus on getting the decay rate estimates. The following are things to keep in mind:

1. filter the noise (low expressed genes could be just noise)
2. I am not using the SPIKE-IN normalization, so I am getting a decay rate in step of half-life.
3. I can do pooling with a hierarchical model to account for noise. I will give a try.

To account for the noise I filter out low expressed genes, and this was done with the ZFPKM package to detect the active genes. The following plot shows the distribution of the genes after filtering. 

```{r amanitingenesexpressed, out.width='70%', fig.cap='distribution expressed genes'}
knitr::include_graphics("../../results/19-01-11-GetDecayRateFromTimeCourse/figures/active_genes-1.png")
```


I fitted a log-linear model to the TPM to get the estimates. The followuing plot shows the bi-variate distribution of the model parameters.

```{r paramsdistribution, fig.cap='model parameters, each dot is a gene'}
knitr::include_graphics("../../results/19-01-11-GetDecayRateFromTimeCourse/figures/get_estimates-1.png")
```

Note that at low expression the $\beta$ estimates tend to be more positve, this can be do to noise, consider filtering genes with alpha > 0.


## 19-01-17

#### Download ORFs and 3' UTR seqs from with biomartR

+ I downloaded the coding and 3' utr sequences for fish, I generated a table with the codon composition.

The data is in: *data/19-01-17-Get-ORFS-UTRS-codon-composition*

```{r listfiles}
list.files("../../data/19-01-17-Get-ORFS-UTRS-codon-composition/sequence-data/")
```



#### Compute CSC

Compute the CSC, this is a way to see if our a-amanitin experiments and
half-life estimation works. It should correlate with the CSC of EMBO 2016.

The following plot displays the distribution of the decay rates

```{r decayrate, fig.cap="decay rate distribution"}
knitr::include_graphics("../../results/19-01-17-EstimateCSC/figures/vis-1.png")
```

This decay rate estimates show a correlation with the old a-amanitin data set.

```{r corold, fig.cap='correlation'}
knitr::include_graphics("../../results/19-01-17-EstimateCSC/figures/cor_with_old_data-1.png")
```


So we expect to see a correlation with the CSC estimates of the previous unpublished a-amanitin data set.

```{r corrCSC}
knitr::include_graphics("../../results/19-01-17-EstimateCSC/figures/csc_cors-1.png")
```



## 19-01-18

+ I created a project specific module to the [codonr package](https://github.com/santiago1234/codonr/blob/master/R/project_mzt_utils.R).
+ I put a function to load the codon composition and the decay rate.

#### Supervised Dimensional Reduction of the Codon Space

I aplied the PLS analysis to the codon space supervised with the decay rate, these PLS component can be used as a fancier version of codon optimality, usefull when doing the model comparison between m6A or miR430 pathways.

```{r pls, fig.cap="PLS components"}
knitr::include_graphics("../../results/19-01-18-ObtainEstimateOfCodonOptimalityPLS/figures/pls_cor_decay-1.png")
```

Here, I also showd that codon optimality affects mRNA level.

```{r plsrna, fig.cap="latent variables affect mRNA level"}
knitr::include_graphics("../../results/19-01-18-ObtainEstimateOfCodonOptimalityPLS/figures/pls_expression-1.png")
```

This analysis will be useful later.

## 19-01-30

#### Predictive model mRNA stability

I trained several predictive models of mRNA stability (decay rate a-amanitin profile) with the codon composition. The top-performing model is an SVM. This model explain about 1/4 of the variation in the mRNA stability.


```{r machinelearning, fig.cap="trained models and performance on a-amanitin data set"}
knitr::include_graphics("../../results/19-01-22-PredictiveModelMrnaStability/figures/results-1.png")
```

Below is a plot  of the predictions in a test set for each model.

```{r predresid, fig.cap="predictions and residuals across models"}
knitr::include_graphics("../../results/19-01-22-PredictiveModelMrnaStability/figures/test_preds-1.png")
```

I used the default tuning parameters for the SVM model, but there is a chance that I can tune over not searched parameters and increase the predictive performance.
