# February

```{r settings_feb, echo=FALSE}
knitr::opts_chunk$set(echo = F)
```

## Building A predictive model of mRNA stability with 3' UTR

### 19-02-01

My goal of the following analysis is to answer the following questions:

+ How well do the 3' UTR sequences (concerning 6-mers) predict mRNA stability?
+ Do they predict better than the codons?
+ What about 3'UTR and codons together?
+ Can we identify regulatory k-mers that drive mRNA stability in the a-amanitin data?
+ Do these sequences affect expression level (genes with the element are more or less expressed)?

To do this i will built the following predictive models

+ UTR model
+ UTR and codons model

Then I will do feature importance analysis to identify potentially regulatory k-mer sequences.


I generated a file with the 6-mer composition *data/19-01-17-Get-ORFS-UTRS-codon-composition/sequence-data/zfish_3utr6mer_composition.csv*.

**I DID NOT FINISH THIS ANALYSIS**

## Profiles of regulatory sequences in RNA-seq time course

The goal is to evaluate the effect of regulatory elements (k-mers, MiR-430, etc.) in the RNA-seq profiles. Ariel suggested to do cumulative plots and also to show the elements that were previously detected.

### 19-02-05

I generated a tidy version of the data with the log2FCs and also a tidy version of the time course (*../../data/19-02-05-FoldChangeData/*). 

Below is a plot of log2FC for some genes, the log2FC was computed with respect to the earliest time point in the time course.

```{r log2FC, out.width='130%', fig.cap='log2 time point / earliest time point'}
knitr::include_graphics("../../data/19-02-05-FoldChangeData/figures/computelog2FC-1.png")
```

I generated a plot of the regulatory effect for the previously identified k-mer sequences in the log2 fold change data.

```{r kmerprofile, fig.cap='k-mer profiles'}
knitr::include_graphics("../../results/19-02-05-PlotElementsProfiles/figures/old_candidate_kmers_profilies-1.png")
```


We can see that only the stabilizing k-mers show strong evidence of affecting mRNA stability. But now that we have the time course we can develop a better model to identify potential regulatory elements. 

## Identifying Regulatory K-mers

We hypothesize that codon optimality has a masking effect; therefore, we need to control for it. Also, we need to control for MiR-430 since we don't need to rediscover it. 

I propose the following model:

$$
logRNA = \alpha_0 + \beta_0logRNA_{t=0} + \beta_1Optimality + \beta_2MiR430 + \beta_3time + \beta_4kmer + \beta_5time\;kmer
$$
more on the parameters:

+ $\beta_0$ is the RNA level at t=0, if we substract from both sides then we are modeling
the log2 fold change. (I may not included and just simply model log fold change).

+ $\beta_1$ This is codon optimality, here I use the 2 PLS components produced in as an indicator of codon optimality in [Supervised Dimensional Reduction of the Codon Space].

+ $\beta_2$ MiR-430 indicator variable

+ $\beta_3$ Time affects expression

+ $\beta_4$ Elements

+ $\beta_5$ More importantly i need to define how the element interacts with time, this interaction may be a polynomial interaction.


Model critic

+ The data in this model suffers from [heteroscedasticity](https://en.wikipedia.org/wiki/Heteroscedasticity), due to transcription, i need to consider this when modeling.

I will fit this model for each 6-mer there will be $2^6$ 6-mers, when modeling I will also do a comparison with the null model to save this as a metric.

### 19-02-06

Added helper functions to the [codonr](https://github.com/santiago1234/codonr/blob/master/R/project_mzt_mdl_RNA_seq_dynamics.R) package for the analysis


### 19-02-07

I used a [horshoe prior](https://discourse.mc-stan.org/t/help-with-lasso-example-in-brms/3987) for the $\beta_s$ in the model [@carvalho2010horseshoe]. I change to a lasso prior, sampling was taking forever.

I will evaluate the k-mer effect with the following Bayesian model:

$$
\begin{aligned}
log2FC &\sim Normal(\mu_i, \sigma_i) \\
\mu_i &= \alpha + \beta_1 t + \beta_2 t^2 + O_i + MiR_i + K_i\\
O_i &= PLS_1(\beta_3 + \beta_4 t) + PLS_2(\beta_5 + \beta_6 t) \\
MiR_i &= M(\beta_7 + \beta_8 t)\\
K_i &= K(\beta_9 + \beta_{10} t)\\
\sigma_i &= \alpha + \gamma_1 t + \gamma_2 t^2 \\
\vec{\beta},\vec{\gamma}  &\sim  Laplace(\lambda) 
\end{aligned}
$$

Variables definition

+ $t$ time in hours
+ $O_i$ codon optimality for gene$_i$: ($PLS_1$ and $PLS_2$ are the PLS components)
+ $M$ number of Mir430 6-mers sites in the 3' UTR
+ $K$ number of k-mer sites in the 3' UTR

The null hypothesis is $\beta_9 = 0$, $\beta_{10} = 0$, which kmers are significant?

I will judge the models (k-mer) bassed on size effect and deviance (waic).

Below is a plot for the response distribution and the relation with some predictors.

```{r mdlkmers2, fig.cap="response and predictors"}
knitr::include_graphics("../../results/19-02-06-DetectRegulatoryKmers/figures/01_viz-1.png")
```


### 19-02-13

I've being runing the model. 

The following plot summarizes the results.


```{r mdres, fig.cap="model summary"}
knitr::include_graphics("../../results/19-02-06-DetectRegulatoryKmers/figures/02_mdlsummary1-1.png")
```


This plot shows the effect of the mid-time point (~6hr) in the x-axis and the interaction
time effect (y-axis). How can we select the top sequences?.

I will create a data frame that for each k-mer has the following information:

+ status: stabilizing (effect positive or negative in both ribo and poly wt, an effect at 6)
+ there are six parameters, for each parameter I will say if 0 is in the interval (significance)
+ columns with the impact (for the six parameters)
+ zygotic program (if the status is significant) is there a positive effect in a-amanitin.


### 19-02-20

I selected the candidate sequences with the following criteria.

They should show a significant effect (95 CI not overlaps 0) in both parameters (k-mer effect and time/k-mer interaction) and both wild type conditions. Most of the resulting k-mers seem to show a stabilizing, only for destabilizing effect I allowed to be significant in at least one condition.

the following plot highlights the candidate sequences.

```{r cand2}
knitr::include_graphics("../../results/19-02-06-DetectRegulatoryKmers/figures/02_showcandidates-1.png")
```

And a hetamp with the effect over time.

```{r cand4, fig.cap="effect over time", , out.width='30%'}
knitr::include_graphics("../../results/19-02-06-DetectRegulatoryKmers/figures/02_candidates_heatmpa-1.png")
```

With this, I have a list of 69 candidates sequences. What are the best k-mers? For this, I will do a feature selection analysis.

## Feature Selection to identify top k-mers sequences

I fit the following model using the LASO, the model was fit independently for each time point and condition.


$$
\begin{aligned}
log_2(\text{fold change}) = \alpha + \beta_1PLS_{1i} + \beta_2PLS_{2i} + 
\beta_3Mir430_{i} +
\sum_{i=4}^{72} \beta_i\;kmer_i + \lambda \sum_{j=1}^{72} |\beta_j|
\end{aligned}
$$

The following plot shows the lasso coefficient path across different values for $\lambda$

```{r lassopath, fig.cap="laso path, l value with best cross validation error is shown in black dotted line", out.width='70%'}
knitr::include_graphics("../../results/19-02-12-TopKmersFeatureSelection/figures/01_lasso_path-1.png")
```

The following plot shows the coefficient for each k-mer that are non zero at the best
$\lambda$ value.

```{r coefslasso, out.width='70%'}
knitr::include_graphics("../../results/19-02-12-TopKmersFeatureSelection/figures/01_coefs_at_six-1.png")
```


Ariel suggested evaluate this k-mers in the other old data sets, including the mzdicer and lna mutants.

Here, I plot the k-mer effect in this data.

```{r crossval, out.width='50%'}
knitr::include_graphics("../../results/19-02-12-TopKmersFeatureSelection/figures/02_marg_efcts-1.png")
```

After this I have identified 5 potentially regulatory k-mers: 

stabalizing `TCGGCG CTCCCG CCTGGG`
destabalizing: `AGACGG, TCCGTA`

Here is plot putting the elements in the same bag:

```{r cumeffect}
knitr::include_graphics("../../results/19-02-12-TopKmersFeatureSelection/figures/02_cum_effect-1.png")
```

